### PR TITLE
i18n(it): Updated islands.mdx and why-astro.mdx

### PR DESCRIPTION
**Pull request by casungo**
#### Description (required)

Updated `islands.mdx` and `why-astro.mdx`

### CODE REVIEW

**Comment by casungo** [comment]
Fixed a typo in the title 😅

**Comment by casungo** [comment]
The failed check is a broken link check. It complaints about /it/reference/directives-reference/#direttive-client not being there because not even the reference folder is created 😄 . I will create it and try to resolve this as soon as I can

File: src/content/docs/it/concepts/islands.mdx
Comment (dreyfus92) [inline src/content/docs/it/concepts/islands.mdx:83]:
```suggestion
Ancora meglio, puoi dire ad Astro esattamente come e quando renderizzare ogni componente. Se quel carosello di immagini è davvero costoso da caricare, puoi allegare una [direttiva client](/it/reference/directives-reference/#client-directives) speciale che dice ad Astro di caricare il carosello solo quando diventa visibile nella pagina. Se l'utente non lo vede mai, non viene mai caricato.
```

**Review by dreyfus92 — COMMENTED** [review]
Amazing work @casungo, I left you a small suggestion 😁

File: src/content/docs/it/concepts/islands.mdx
Comment (casungo) [reply src/content/docs/it/concepts/islands.mdx:83]: I addressed this in [this comment](https://github.com/withastro/docs/pull/6837#issuecomment-1934898705), do you think it should not be translated in the meantime?

Comment (dreyfus92) [reply src/content/docs/it/concepts/islands.mdx:83]: Leave it as it is. Every time we merge something into the repo a maintainer should update the PR in order to run all checks to safely merge. If I do it or someone else should first merge the PR who has no problems to be merged.

Comment (delucis) [reply src/content/docs/it/concepts/islands.mdx:83]: The change suggested by @dreyfus92 is required before merging this PR. Then when the directives reference page is updated, that PR can include an update also to this page to use `#direttive-client`.

**Review by delucis — CHANGES_REQUESTED** [review]
Thanks for the PR @casungo — looks good to me. We do need to fix the broken link before merging though as @dreyfus92 suggested.

**Review by delucis — APPROVED** [review]

**Pull request merged**
